CONTRIBUTING.md

```diff
@@ -0,0 +1,119 @@
+8. Core members will review the pull request and provide feedback when necessary
+   - If a PR is too large, you may be asked to break it down into multiple smaller-scoped PRs
+9. Once the PR is approved by a core member, it will be merged
+10. Celebrate! Your code is released 🎈🎉🍻
+
+## Table of Contents
+
+- [Opening an Issue](#opening-an-issue)
+  - [Grammar](#grammar)
+- [Pull Request Guidelines](#pull-request-guidelines)
+  - [Documentation](#documentation)
+  - [Testing](#testing)
+  - [Code Style](#code-style)
+  - [Git Commit](#git-commit)
+
+## Opening an Issue
+
+The title of a Bug or Enhancement should clearly indicate what is broken or desired. Use the description to explain possible solutions or add details and (especially for Enhancements) explain _how_ or _why_ the issue is broken or desired. Follow the template!
+
+### Grammar
+
+While quibbling about grammar in issue titles may seem a bit pedantic, adhering to some simple rules can make it much easier to understand a Bug or an Enhancement from the title alone. For example, is the title **"Browsers should support blinking text"** a bug or a feature request?
+
+- Enhancements: The title should be an imperative statement of how things should be. **"Add support for blinking text"**
+- Bugs: The title should be a declarative statement of how things are. **"Text does not blink"**
+
+## Pull Request Guidelines
+
+Pull requests must include code documentation, tests, follow code style and commits format.
+
+### Documentation
+
+All methods, functions and object structures should be documented following [JSDoc](https://jsdoc.app/index.html) style comments.
+
+### Testing
+
+We take testing very seriously, all code changes must include unit, integration and end-to-end tests.
+
+- **Unit**: Tests at the file level with mocked external requests
+- **Integration**: Tests at the application level with mocked I/O requests
+- **End-to-end**: Tests the application in a system
+
+### Code Style
+
+Code style is enforced by [linters](https://eslint.org). Use `npm run test:eslint` to verify that your code is beautiful, too!
+We highly discourage disabling eslint rules.
+Unless there is an exceptional use case, we may request additional changes to your PR.
+
+### Git Commit
+
+As part of the build process, commits are run through [conventional changelog](https://github.com/conventional-changelog/conventional-changelog) to generate the changelog. Please adhere to the following guidelines when formatting your commit messages.
+
+#### Commit Message Format
+
+Each commit message consists of a **header**, a **body** and a **footer**. The header has a special format that includes a **type**, a **scope** and a **subject**:
+
+    <type>(<scope>): <subject>
+    <BLANK LINE>
+    <body>
+    <BLANK LINE>
+    <footer>
+
+The **header** is mandatory and the scope of the header is optional.
```

Comment:
I know this is what the other contribution guides have too, but should the scope be optional?
@adamweeks @akoushke @taymoork2 @InteractiveTimmy

Reply:
what's inside the parentheses? I don't think that should be optional. semantic looks at that and adds it the issues and releases. so it make our changelog, issue and release more informative.

Reply:
believe scope being optional is intentional

https://www.conventionalcommits.org/en/v1.0.0/

Reply:
Yes, it's what is in parenthesis and what determines where the change happens in the CHANGELOG. That's why I'm bringing it up

Reply:
true, it is optional but I'm saying its more informative